keyboards/melgeek/mach80/rev1/rev1.c

```diff
@@ -0,0 +1,193 @@
+		{0, 24},  {12, 24},  {24, 24}, {36, 24}, {48, 24}, {60, 24}, {72, 24}, {84, 24}, {96, 24}, {108, 24}, {120, 24}, {132, 24}, {144, 24}, {156, 24}, {168, 24}, {180, 24}, {192, 24},                       //17
+		{0, 36},  {12, 36},  {24, 36}, {36, 36}, {48, 36}, {60, 36}, {72, 36}, {84, 36}, {96, 36}, {108, 36}, {120, 36}, {132, 36}, {144, 36}, {156, 36}, {168, 36},                                             //15
+		{0, 48},  {12, 48},  {24, 48}, {36, 48}, {48, 48}, {60, 48}, {72, 48}, {84, 48}, {96, 48}, {108, 48}, {120, 48}, {132, 48}, {144, 48}, {156, 48}, {168, 48}, {180, 48}, {192, 48},                       //17
+		{0, 60},  {12, 60},  {24, 60}, {36, 60}, {48, 60}, {60, 60}, {72, 60}, {84, 60}, {96, 60}, {108, 60}, {120, 60}, {132, 60}, {144, 60}                                                                    //13
+	}, {          
+	       4,         4,         4,        4,        4,        4,        4,         4,         4,         4,         4,         4,         4,         4,         4,         4,       
+		   4,         4,         4,        4,        4,        4,        4,         4,         4,         4,         4,         4,         4,         4,         4,         4,         4,         4,         4,
+		   4,         4,         4,        4,        4,        4,        4,         4,         4,         4,         4,         4,         4,         4,         4,         4,         4,
+		   4,         4,         4,        4,        4,        4,        4,         4,         4,         4,         4,         4,         4,         4,         4,         
+		   4,         4,         4,        4,        4,        4,        4,         4,         4,         4,         4,         4,         4,         4,         4,         4,         4,
+		   4,         4,         4,        4,        4,        4,        4,         4,         4,         4,         4,         4,         4,
+	}
+};
+
+const is31_led g_is31_indicator_leds[3] = {
+    {0, CS6_SW1, CS5_SW1, CS4_SW1}, /* RGB107 */
+    {0, CS6_SW2, CS5_SW2, CS4_SW2}, /* RGB108 */
+    {0, CS6_SW3, CS5_SW3, CS4_SW3}, /* RGB109 */
+};
+
+bool led_update_kb(led_t led_state) {
+    if (led_update_user(led_state)) {
+        if (led_state.caps_lock) {
+            IS31FL3741_set_pwm_buffer(&g_is31_indicator_leds[2], 0x00, 0x00, 0xff);
+        } else {
+            IS31FL3741_set_pwm_buffer(&g_is31_indicator_leds[2], 0x00, 0x00, 0x00);
+        }
+
+        if (led_state.num_lock) {
+            IS31FL3741_set_pwm_buffer(&g_is31_indicator_leds[1], 0xff, 0x00, 0x00);
+        } else {
+            IS31FL3741_set_pwm_buffer(&g_is31_indicator_leds[1], 0x00, 0x00, 0x00);
+        }
+
+        if (led_state.scroll_lock) {
+            IS31FL3741_set_pwm_buffer(&g_is31_indicator_leds[0], 0x00, 0xff, 0x00);
+        } else {
+            IS31FL3741_set_pwm_buffer(&g_is31_indicator_leds[0], 0x00, 0x00, 0x00);
+        }
+    }
+    return true;
+}
```

Comment:
This is a really nonstandard implementation of indicators using the low-level driver API directly. Maybe a better way would be to add those indicator LEDs into the RGB matrix too, mark them with `LED_FLAG_INDICATOR`, then implement `rgb_matrix_indicators_advanced_kb()` to control those LEDs using the public RGB Matrix API.

One potential issue with that, however, is that there is no way to disable the `rgb_matrix_indicators_advanced_kb()` from the keymap level; although overriding the state of the same LEDs in `rgb_matrix_indicators_advanced_user()` would work, reverting those LEDs to display the regular RGB matrix effect would not be possible, unless the keyboard level implementation of `rgb_matrix_indicators_advanced_kb()` exposes some nonstandard ways to do that. However, it's unlikely that someone would want to include such indicator LEDs into the animations, so that restriction might still be acceptable.

Another issue is that if you choose to implement `rgb_matrix_indicators_kb()`, you would get somewhat inefficient code (the `rgb_matrix_indicators_kb()` function might be invoked multiple times for the same frame), while if you implement `rgb_matrix_indicators_advanced_kb()`, then the user would not be able to use a simple `rgb_matrix_indicators_user()` to override it — only `rgb_matrix_indicators_user_advanced()` would work.